src/System.Private.CoreLib/shared/System/Threading/Tasks/AsyncCausalityTracer.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !FEATURE_COMINTEROP
+
+using System.Diagnostics;
+
+namespace System.Threading.Tasks
+{
+    internal enum CausalityTraceLevel
+    {
+        Required = 0,
+        Important = 1,
+        Verbose = 2,
+    }
+
+    internal enum AsyncCausalityStatus
+    {
+        Started = 0,
+        Completed = 1,
+        Canceled = 2,
+        Error = 3,
+    }
+
+    internal enum CausalityRelation
+    {
+        AssignDelegate = 0,
+        Join = 1,
+        Choice = 2,
+        Cancel = 3,
+        Error = 4,
+    }
+
+    internal enum CausalitySynchronousWork
+    {
+        CompletionNotification = 0,
+        ProgressNotification = 1,
+        Execution = 2,
+    }
+
+    //
+    // Empty implementation of AsyncCausality events
+    //
+    internal static class AsyncCausalityTracer
+    {
+        public static readonly bool LoggingOn = false;
```

Comment:
This can be const.

Reply:
In theory yes but we end up with a lot of C# Unreachable Code warnings, it that ok?

Reply:
Ok, we do not like warnings. Maybe property returning `false` would work better - the codegen can always see that it is false, without triggering ctors, etc.

Reply:
If you make it a property, linker can recognize property returning constant too and it is easier to do right.

Properties returning constant cannot be overridden by reflection or other means. It makes them easy for optimizers to reason about.

Static fields are much harder for optimizers to reason about because of reflection, cctor triggers, static fields set by debugger, etc.